OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -53,6 +53,7 @@ public enum UnitStance { HoldFire, ReturnFire, Defend, AttackAnything }
 	public class AutoTarget : ConditionalTrait<AutoTargetInfo>, INotifyIdle, INotifyDamage, ITick, IResolveOrder, ISync
 	{
 		readonly AttackBase[] attackBases;
+		readonly IEnumerable<AttackBase> activeAttackBases;
```

Comment:
You pull out an `IEnumerable` for this, but then also replace the enumerable for `attackFollows` with an explicit loop.  Why does this case not justify an explicit loop?